plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/geoip/GeoIpProcessor.java

```diff
@@ -68,13 +71,21 @@ public void execute(Data data) {
             throw new RuntimeException(e);
         }
 
-        final Map<String, Object> geoData;
+        Map<String, Object> geoData;
         switch (dbReader.getMetadata().getDatabaseType()) {
             case "GeoLite2-City":
-                geoData = retrieveCityGeoData(ipAddress);
+                try {
+                    geoData = retrieveCityGeoData(ipAddress);
+                } catch (Exception e) {
+                    geoData = Collections.singletonMap("error", ExceptionsHelper.detailedMessage(e));
```

Comment:
I wonder, should this tagging be somewhere more generic? seems like it will be needed pretty much everywhere?